2.Project2/datasource.h

```diff
@@ -34,16 +34,26 @@ class DataSource : public QObject
     void preItemRemoved(int index);
     void postItemRemoved();
 
+    void requestCompleted();
 private slots:
     void dataReadyRead();
     void dataReadFinished();
 
+    void makeRequest();
 public slots:
 private :
     QNetworkAccessManager * mNetManager;
     QNetworkReply * mNetReply;
     QByteArray * mDataBuffer;
     QList<Joke*> mJokes;
+
+    typedef enum : unsigned int {
```

Comment:
Consider using C++ scoped enumerations: https://en.cppreference.com/w/cpp/language/enum

Reply:
Okay, I have updated the enumeration to scoped type.